src/nvme/mi.c

```diff
@@ -616,3 +617,28 @@ void nvme_mi_close_ctrl(nvme_mi_ctrl_t ctrl)
 {
 	free(ctrl);
 }
+
+char *nvme_mi_endpoint_desc(nvme_mi_ep_t ep)
+{
+	char *s = NULL, *ts = NULL;
+	int len = 0;
+
+	if (ep->transport->desc_ep)
+		ts = ep->transport->desc_ep(ep);
+
+	len = 30;
+	len += strlen(ep->transport->name);
+	if (ts)
+		len += strlen(ts);
+
+	s = malloc(len);
+	if (s) {
+		if (ts)
+			snprintf(s, len, "%s: %s", ep->transport->name, ts);
```

Comment:
the `strlen` and `malloc` dance looks a bit awkward. `asprintf` avoids this altogether